app/Repositories/CacheUserRepository.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Gladiator\Repositories;
+
+use Gladiator\Models\User;
+use Illuminate\Support\Facades\Cache;
+
+class CacheUserRepository implements UserRepositoryInterface
+{
+    /**
+     * UserRepositoryInterface instance.
+     *
+     * @var \Gladiator\Repositories\UserRepositoryInterface
+     */
+    protected $database;
```

Comment:
Another nitpick, but this should probably just be `protected $repository` since in theory you could nest any "conforming" repository in here that matches the contract.